lib/possible_bipartition.rb

```diff
@@ -1,4 +1,41 @@
+# I tried my best! All the test pass except for number 4. I know that the problem
+# is that I should be moving through the dislikes as they appear instead of just
+# linearly. However, for whatever when I tried using a stack and popping off the 
+# values, it didnt alter my stack--meaning the stack grew in length and I was 
+# trapped in a loop.  
```

Comment:
Yeah, you are going to need a stack or queue to traverse the graph so that you can keep consistent that you going toggling between the two "colors" or groups while on a path. 

Your code here isn't traversing a graph, but instead iterating through the list of nodes.  Take a look at either my breadth-first-search or depth-first-search solutions.